packages/nimble-components/src/table-column/tests/table-column.stories.ts

```diff
@@ -0,0 +1,416 @@
+import { html, ref, when } from '@microsoft/fast-element';
+import type { Meta, StoryObj } from '@storybook/html';
+import { withXD } from 'storybook-addon-xd-designs';
+import {
+    createUserSelectedThemeStory,
+    usageWarning
+} from '../../utilities/tests/storybook';
+import { ExampleSortType } from './types';
+import { Table, tableTag } from '../../table';
+import { TableColumnSortDirection } from '../../table/types';
+import { iconUserTag } from '../../icons/user';
+import { iconCommentTag } from '../../icons/comment';
+import { tableColumnTextTag } from '../text';
+
+interface CommonTableArgs {
+    idFieldName: undefined;
+    tableRef: Table;
+    updateData: (args: CommonTableArgs) => void;
+}
+
+const simpleData = [
+    {
+        firstName: 'Ralph',
+        lastName: 'Wiggum',
+        favoriteColor: 'Rainbow',
+        quote: "I'm in danger!"
+    },
+    {
+        firstName: 'Milhouse',
+        lastName: 'Van Houten',
+        favoriteColor: 'Crimson',
+        quote: "Not only am I not learning, I'm forgetting stuff I used to know!"
+    },
+    {
+        firstName: 'Ned',
+        lastName: 'Flanders',
+        favoriteColor: 'Taupe',
+        quote: 'Hi diddly-ho neighbor!'
+    },
+    {
+        firstName: 'Maude',
+        lastName: 'Flanders',
+        favoriteColor: 'Taupe',
+        quote: "Neddy doesn't believe in insurance. He considers it a form of gambling."
+    },
+    {
+        firstName: 'Rod',
+        lastName: 'Flanders',
+        favoriteColor: 'Taupe',
+        quote: 'Lies make baby Jesus cry.'
+    },
+    {
+        firstName: 'Todd',
+        lastName: 'Flanders',
+        favoriteColor: 'Taupe',
+        quote: 'Dad, should I poke Rod with a sharp thing like the mouse did?'
+    }
+] as const;
+
+const idFieldName = 'firstName';
+
+const overviewText = `This page contains information about configuring the columns of a \`nimble-table\`. 
+See the **Table** story for information about configuring the table itself.`;
+
+const metadata: Meta<CommonTableArgs> = {
+    title: 'Table Columns',
```

Comment:
What do you think about having this render under Table in the navigation tree? Kinda how Menu Item is under menu

I kinda like the organization of `Table/Column Configuration`:
![image](https://user-images.githubusercontent.com/1588923/224375102-4c1e4523-c8f8-4027-9ed1-5f8a1ef32b48.png)

Reply:
I like the idea but I can't figure out how to get it to work. If I use the `Table/Column Configuration` naming as described in [Storybook docs](https://storybook.js.org/docs/react/writing-stories/naming-components-and-hierarchy#grouping) I end up with a separate top-level section (sibling to Tokens and Tests). I did briefly see the purple folder layout in your screenshot but that only seems to appear after a hot reload on localhost; if I refresh the page after that I get the separate section.

<img width="215" alt="image" src="https://user-images.githubusercontent.com/10500124/224448435-8ad6db51-b3fc-44b3-b83b-bbaaa05ad0fa.png">

I think a separate section for the table is worse than having multiple pages for column configuration so would vote to keep things as-is if we can't figure out how to get the layout in your screenshot to work reliably.

Reply:
I'm actually kinda feeling the table as a separate section. It's a sufficiently complex component with potentially dozen of interrelated elements to be aware of and lots of nuance.
But it's also a comparatively minor organization point and not worth blocking this PR. I'd rather the PR go in sooner so we can start capturing docs as features are merged and refine the organization more later.